frontends/web/src/routes/account/send/send.tsx

```diff
@@ -338,33 +339,9 @@ class Send extends Component<Props, State> {
       if (updateFiat) {
         this.convertToFiat(result.amount.amount);
       }
-    } else {
-      const errorCode = result.errorCode;
-      switch (errorCode) {
-      case 'invalidAddress':
-        this.setState({ addressError: this.props.t('send.error.invalidAddress') });
-        break;
-      case 'invalidAmount':
-      case 'insufficientFunds':
-        this.setState({
-          amountError: this.props.t(`send.error.${errorCode}`),
-          proposedFee: undefined,
-        });
-        break;
-      case 'feeTooLow':
-        this.setState({ feeError: this.props.t('send.error.feeTooLow') });
-        break;
-      case 'feesNotAvailable':
-        this.setState({ feeError: this.props.t('send.error.feesNotAvailable') });
-        break;
-      default:
-        this.setState({ proposedFee: undefined });
-        if (errorCode) {
-          this.unregisterEvents();
-          alertUser(errorCode, { callback: this.registerEvents });
-        }
-      }
-      this.setState({ isUpdatingProposal: false });
+    } else if (result.errorCode) {
+      const errorHandling = txProposalErrorHandling(this.registerEvents, this.unregisterEvents, result.errorCode);
+      this.setState({ ...errorHandling, isUpdatingProposal: false });
     }
```

Comment:
If I read correctly now `this.setState({ isUpdatingProposal: false });` only runs if result.errorCode is truthy. Before this change the state was always updated after the switch.

Reply:
I guess it's the same question with [the other one](https://github.com/digitalbitbox/bitbox-wallet-app/pull/2421/files/4f8053f1e227d809b0c954212ebd74f12d61f0b7#r1427876867). Indeed I should've kept the else, not else if! 

Thanks:)